{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# geofetch tutorial for processed data\n",
    "\n",
    "The [GSE150868 data set](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE150868) has about 1.2 gb of processed data that contains 57 Supplementary files, so it's a quick download for a test case. Let's take a quick peek at the geofetch version:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "geofetch 1.0.0\n"
     ]
    }
   ],
   "source": [
    "geofetch --version"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To see your CLI options, invoke `geofetch -h`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: geofetch [-h] [-V] -i INPUT [-n NAME] [-m METADATA_ROOT]\n",
      "                [-u METADATA_FOLDER] [--just-metadata] [-r] [--acc-anno]\n",
      "                [--use-key-subset] [-x] [--config-template CONFIG_TEMPLATE]\n",
      "                [-k SKIP] [-p] [--stored-in {all,samples,series}]\n",
      "                [--filter FILTER] [--filter-size FILTER_SIZE] [-g GEO_FOLDER]\n",
      "                [-b BAM_FOLDER] [-f FQ_FOLDER] [-P PIPELINE_INTERFACES]\n",
      "                [--silent] [--verbosity V] [--logdev]\n",
      "\n",
      "Automatic GEO and SRA data downloader\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  -V, --version         show program's version number and exit\n",
      "  -i INPUT, --input INPUT\n",
      "                        required: a GEO (GSE) accession, or a file with a list\n",
      "                        of GSE numbers\n",
      "  -n NAME, --name NAME  Specify a project name. Defaults to GSE number\n",
      "  -m METADATA_ROOT, --metadata-root METADATA_ROOT\n",
      "                        Specify a parent folder location to store metadata.\n",
      "                        The project name will be added as a subfolder\n",
      "                        [Default: $SRAMETA:]\n",
      "  -u METADATA_FOLDER, --metadata-folder METADATA_FOLDER\n",
      "                        Specify an absolute folder location to store metadata.\n",
      "                        No subfolder will be added. Overrides value of\n",
      "                        --metadata-root [Default: Not used (--metadata-root is\n",
      "                        used by default)]\n",
      "  --just-metadata       If set, don't actually run downloads, just create\n",
      "                        metadata\n",
      "  -r, --refresh-metadata\n",
      "                        If set, re-download metadata even if it exists.\n",
      "  --acc-anno            Also produce annotation sheets for each accession, not\n",
      "                        just for the whole project combined\n",
      "  --use-key-subset      Use just the keys defined in this module when writing\n",
      "                        out metadata.\n",
      "  -x, --split-experiments\n",
      "                        Split SRR runs into individual samples. By default,\n",
      "                        SRX experiments with multiple SRR Runs will have a\n",
      "                        single entry in the annotation table, with each run as\n",
      "                        a separate row in the subannotation table. This\n",
      "                        setting instead treats each run as a separate sample\n",
      "  --config-template CONFIG_TEMPLATE\n",
      "                        Project config yaml file template.\n",
      "  -k SKIP, --skip SKIP  Skip some accessions. [Default: no skip].\n",
      "  -p, --processed       Download processed data [Default: download raw data].\n",
      "  --stored-in {all,samples,series}\n",
      "                        Optional: {processed} Specify GEO location where\n",
      "                        processed data is stored and has to be downloaded.\n",
      "                        Processed data is stored in samples and experiments.\n",
      "                        Allowable values are: samples, series or both (all).\n",
      "                        [Default: all]\n",
      "  --filter FILTER       Optional: {processed} Filter regex for processed\n",
      "                        filenames [Default: None].\n",
      "  --filter-size FILTER_SIZE\n",
      "                        Optional: {processed} Filter size for processed files\n",
      "                        that are stored as sample repository [Default: None].\n",
      "                        Supported input formats : 12B, 12KB, 12MB, 12GB\n",
      "  -g GEO_FOLDER, --geo-folder GEO_FOLDER\n",
      "                        Optional: {processed} Specify a location to store\n",
      "                        processed GEO files [Default: $GEODATA:]\n",
      "  -b BAM_FOLDER, --bam-folder BAM_FOLDER\n",
      "                        Optional: Specify folder of bam files. Geofetch will\n",
      "                        not download sra files when corresponding bam files\n",
      "                        already exist. [Default: $SRABAM:]\n",
      "  -f FQ_FOLDER, --fq-folder FQ_FOLDER\n",
      "                        Optional: Specify folder of fastq files. Geofetch will\n",
      "                        not download sra files when corresponding fastq files\n",
      "                        already exist. [Default: $SRAFQ:]\n",
      "  -P PIPELINE_INTERFACES, --pipeline_interfaces PIPELINE_INTERFACES\n",
      "                        Optional: Specify one or more filepaths to pipeline\n",
      "                        interface yaml files. These will be added to the\n",
      "                        project config file to make it immediately compatible\n",
      "                        with looper. [Default: null]\n",
      "  --silent              Silence logging. Overrides verbosity.\n",
      "  --verbosity V         Set logging level (1-5 or logging module level name)\n",
      "  --logdev              Expand content of logging message format.\n"
     ]
    }
   ],
   "source": [
    "geofetch -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calling geofetch will do 4 tasks: \n",
    "\n",
    "1. download all or filtered processed files from `GSE#####` into your geo folder.\n",
    "2. download all metadata from GEO and store in your metadata folder.\n",
    "2. produce a PEP-compatible sample table, `PROJECT_NAME_sample_processed.csv` and `PROJECT_NAME_series_processed.csv`, in your metadata folder.\n",
    "3. produce a PEP-compatible project configuration file, `PROJECT_NAME_sample_processed.yaml` and `PROJECT_NAME_series_processed.yaml`, in your metadata folder.\n",
    "\n",
    "Complete details about geofetch outputs is cataloged in the [metadata outputs reference](metadata_output.md)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download the data\n",
    "\n",
    "First, create the metadata for processed data (by adding --processed and --just-metadata):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Metadata folder: /home/bnt4me/Virginia/for_docs/bright_test\n",
      "Trying GSE150868 (not a file) as accession...\n",
      "Skipped 0 accessions. Starting now.\n",
      "\u001b[38;5;228mProcessing accession 1 of 1: 'GSE150868'\u001b[0m\n",
      "Found previous GSE file: /home/bnt4me/Virginia/for_docs/bright_test/GSE150868_GSE.soft\n",
      "Found previous GSM file: /home/bnt4me/Virginia/for_docs/bright_test/GSE150868_GSM.soft\n",
      "\u001b[38;5;242mFile /home/bnt4me/Virginia/for_docs/bright_test/GSE150868_file_list.txt exists.\u001b[0m\n",
      "Total number of processed SAMPLES files found is: 35\n",
      "Total number of processed SERIES files found is: 2\n",
      "Finished processing 1 accession(s)\n",
      "Expanding metadata list...\n",
      "Unifying and saving of metadata... \n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_sample_processed.csv has been saved successfully\u001b[0m\n",
      "  Config file: /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_sample_processed.yaml\n",
      "Unifying and saving of metadata... \n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_series_processed.csv has been saved successfully\u001b[0m\n",
      "  Config file: /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_series_processed.yaml\n"
     ]
    }
   ],
   "source": [
    "geofetch -i GSE150868 --processed -n bright_test --just-metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bright_test_annotation_sample_processed.csv   GSE150868_file_list.txt\n",
      "bright_test_annotation_sample_processed.yaml  GSE150868_GSE.soft\n",
      "bright_test_annotation_series_processed.csv   GSE150868_GSM.soft\n",
      "bright_test_annotation_series_processed.yaml\n"
     ]
    }
   ],
   "source": [
    "ls bright_test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `.soft` files are the direct output from GEO, which contain all the metadata as stored by GEO, for both the experiment (`_GSE`) and for the individual samples (`_GSM`). Geofetch also produces a `csv` file with the SRA metadata. The filtered version (ending in `_filt`) would contain only the specified subset of the samples if we didn't request them all, but in this case, since we only gave an accession, it is identical to the complete file. Additionally, file_list.txt is downloaded, that contains information about size, type and creation date of all sample files.\n",
    "\n",
    "Finally, there are the 2 files that make up the PEP: the `_config.yaml` file and the `_annotation.csv` file (for samples and series). Let's see what's in these files now."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pep_version: 2.0.0\n",
      "project_name: bright_test\n",
      "sample_table: /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_sample_processed.csv\n",
      "\n",
      "sample_modifiers:\n",
      "  append:\n",
      "    output_file_path: FILES\n",
      "  derive:\n",
      "    attributes: [output_file_path]\n",
      "    sources:\n",
      "      FILES: /{SRA}/{sample_name}\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat bright_test/bright_test_annotation_sample_processed.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are two important things to note in his file: First, see in the PEP that `sample_table` points to the csv file produced by geofetch. Second: output_file_path is location of all the files.\n",
    "\n",
    "Now let's look at the first 100 characters of the csv file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSE,Sample_title,Sample_geo_accession,Sample_status,Sample_submission_date,Sample_last_update_date,S\n",
      "GSE150868,11244,GSM4559921,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapien\n",
      "GSE150868,22412,GSM4559922,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapien\n",
      "GSE150868,23207,GSM4559923,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapien\n",
      "GSE150868,25963,GSM4559924,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapien\n",
      "GSE150868,26774,GSM4559925,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapien\n",
      "GSE150868,26996,GSM4559926,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapien\n",
      "GSE150868,04H055,GSM4559927,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,04H112,GSM4559928,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,05H111,GSM4559929,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,06H133,GSM4559930,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,06H143,GSM4559931,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,07H005,GSM4559932,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,07H009,GSM4559933,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,07H062,GSM4559934,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,07H158,GSM4559935,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,08H082,GSM4559936,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,08H138,GSM4559937,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,09H083,GSM4559938,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,09H115,GSM4559939,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,10H166,GSM4559940,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,11H008,GSM4559941,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,11H027,GSM4559942,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,11H142,GSM4559943,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,11H160,GSM4559944,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,12H010,GSM4559945,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,13H073,GSM4559946,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,13H110,GSM4559947,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,14H001,GSM4559948,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,14H007,GSM4559949,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapie\n",
      "GSE150868,E170001,GSM4559950,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapi\n",
      "GSE150868,E180007,GSM4559951,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapi\n",
      "GSE150868,E180013,GSM4559952,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapi\n",
      "GSE150868,E180015,GSM4559953,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapi\n",
      "GSE150868,E180029,GSM4559954,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapi\n",
      "GSE150868,E180117,GSM4559955,Public on Feb 06 2022,May 19 2020,Feb 07 2022,SRA,1,AML cells,Homo sapi\n"
     ]
    }
   ],
   "source": [
    "cut -c -100 bright_test/bright_test_annotation_sample_processed.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's download the actual data. This time we will will be downloading data from the [GSE165049 data set](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE165049) .\n",
    "Let's additionally add few arguments:\n",
    "- _geo-folder_ (required) - path to the location where processed files have to be saved\n",
    "- _filter_ argument, to download only _bed_ files  (--filter \".Bed.gz$\") \n",
    "- _stored-in_ argument, to download files only from sample location (--stored-in samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Metadata folder: /home/bnt4me/Virginia/for_docs/bright_test\n",
      "Trying GSE165049 (not a file) as accession...\n",
      "Skipped 0 accessions. Starting now.\n",
      "\u001b[38;5;228mProcessing accession 1 of 1: 'GSE165049'\u001b[0m\n",
      "--2022-02-21 22:50:41--  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?targ=gse&acc=GSE165049&form=text&view=full\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 2607:f220:41e:4290::110, 130.14.29.110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|2607:f220:41e:4290::110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [geo/text]\n",
      "Saving to: ‘/home/bnt4me/Virginia/for_docs/bright_test/GSE165049_GSE.soft’\n",
      "\n",
      "/home/bnt4me/Virgin     [ <=>                ]   4.48K  --.-KB/s    in 0s      \n",
      "\n",
      "2022-02-21 22:50:42 (49.6 MB/s) - ‘/home/bnt4me/Virginia/for_docs/bright_test/GSE165049_GSE.soft’ saved [4587]\n",
      "\n",
      "--2022-02-21 22:50:42--  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?targ=gsm&acc=GSE165049&form=text&view=full\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 2607:f220:41e:4290::110, 130.14.29.110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|2607:f220:41e:4290::110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [geo/text]\n",
      "Saving to: ‘/home/bnt4me/Virginia/for_docs/bright_test/GSE165049_GSM.soft’\n",
      "\n",
      "/home/bnt4me/Virgin     [              <=>   ] 250.79K  64.5KB/s    in 3.9s    \n",
      "\n",
      "2022-02-21 22:50:47 (64.5 KB/s) - ‘/home/bnt4me/Virginia/for_docs/bright_test/GSE165049_GSM.soft’ saved [256813]\n",
      "\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:47--  ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE165nnn/GSE165049/suppl/filelist.txt\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/bright_test/GSE165049_file_list.txt’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/series/GSE165nnn/GSE165049/suppl ... done.\n",
      "==> SIZE filelist.txt ... 4684\n",
      "==> EPSV ... done.    ==> RETR filelist.txt ... done.\n",
      "Length: 4684 (4.6K) (unauthoritative)\n",
      "\n",
      "filelist.txt        100%[===================>]   4.57K  --.-KB/s    in 0s      \n",
      "\n",
      "2022-02-21 22:50:49 (962 MB/s) - ‘/home/bnt4me/Virginia/for_docs/bright_test/GSE165049_file_list.txt’ saved [4684]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "Total number of processed SAMPLES files found is: 57\n",
      "\u001b[32mTotal number of files after filter is: 31 \u001b[0m\n",
      "Total number of processed SERIES files found is: 0\n",
      "\u001b[32mTotal number of files after filter is: 0 \u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:49--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025270/suppl/GSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025270/suppl ... done.\n",
      "==> SIZE GSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz ... 1287747\n",
      "==> EPSV ... done.    ==> RETR GSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 1287747 (1.2M) (unauthoritative)\n",
      "\n",
      "GSM5025270_CBS-u2-K 100%[===================>]   1.23M  5.26MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:50:50 (5.26 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz’ saved [1287747]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:50--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025271/suppl/GSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025271/suppl ... done.\n",
      "==> SIZE GSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz ... 753211\n",
      "==> EPSV ... done.    ==> RETR GSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 753211 (736K) (unauthoritative)\n",
      "\n",
      "GSM5025271_CBS-u2-R 100%[===================>] 735.56K  1.87MB/s    in 0.4s    \n",
      "\n",
      "2022-02-21 22:50:51 (1.87 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz’ saved [753211]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:52--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025272/suppl/GSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025272/suppl ... done.\n",
      "==> SIZE GSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz ... 989413\n",
      "==> EPSV ... done.    ==> RETR GSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 989413 (966K) (unauthoritative)\n",
      "\n",
      "GSM5025272_MYC-3CBS 100%[===================>] 966.22K  2.10MB/s    in 0.5s    \n",
      "\n",
      "2022-02-21 22:50:54 (2.10 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz’ saved [989413]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:54--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025273/suppl/GSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025273/suppl ... done.\n",
      "==> SIZE GSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz ... 626217\n",
      "==> EPSV ... done.    ==> RETR GSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 626217 (612K) (unauthoritative)\n",
      "\n",
      "GSM5025273_CBS-u2-K 100%[===================>] 611.54K  3.00MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:50:56 (3.00 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz’ saved [626217]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:56--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025274/suppl/GSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025274/suppl ... done.\n",
      "==> SIZE GSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz ... 629612\n",
      "==> EPSV ... done.    ==> RETR GSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 629612 (615K) (unauthoritative)\n",
      "\n",
      "GSM5025274_CBS-u2-R 100%[===================>] 614.86K  3.52MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:50:57 (3.52 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz’ saved [629612]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:58--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025275/suppl/GSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025275/suppl ... done.\n",
      "==> SIZE GSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz ... 496663\n",
      "==> EPSV ... done.    ==> RETR GSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 496663 (485K) (unauthoritative)\n",
      "\n",
      "GSM5025275_MYC-3CBS 100%[===================>] 485.02K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:50:58 (3.46 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz’ saved [496663]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:50:59--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025276/suppl/GSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025276/suppl ... done.\n",
      "==> SIZE GSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz ... 1080189\n",
      "==> EPSV ... done.    ==> RETR GSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 1080189 (1.0M) (unauthoritative)\n",
      "\n",
      "GSM5025276_WT-CTCF- 100%[===================>]   1.03M  4.59MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:50:59 (4.59 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz’ saved [1080189]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:00--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025277/suppl/GSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025277/suppl ... done.\n",
      "==> SIZE GSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz ... 634336\n",
      "==> EPSV ... done.    ==> RETR GSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 634336 (619K) (unauthoritative)\n",
      "\n",
      "GSM5025277_HOTTIP-K 100%[===================>] 619.47K  2.30MB/s    in 0.3s    \n",
      "\n",
      "2022-02-21 22:51:01 (2.30 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz’ saved [634336]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:01--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025278/suppl/GSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025278/suppl ... done.\n",
      "==> SIZE GSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz ... 807443\n",
      "==> EPSV ... done.    ==> RETR GSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 807443 (789K) (unauthoritative)\n",
      "\n",
      "GSM5025278_WT-RAD21 100%[===================>] 788.52K  1.80MB/s    in 0.4s    \n",
      "\n",
      "2022-02-21 22:51:02 (1.80 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz’ saved [807443]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:03--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025279/suppl/GSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025279/suppl ... done.\n",
      "==> SIZE GSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz ... 544322\n",
      "==> EPSV ... done.    ==> RETR GSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 544322 (532K) (unauthoritative)\n",
      "\n",
      "GSM5025279_HOTTIP-K 100%[===================>] 531.56K  1.41MB/s    in 0.4s    \n",
      "\n",
      "2022-02-21 22:51:04 (1.41 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz’ saved [544322]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:04--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025280/suppl/GSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025280/suppl ... done.\n",
      "==> SIZE GSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz ... 400787\n",
      "==> EPSV ... done.    ==> RETR GSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 400787 (391K) (unauthoritative)\n",
      "\n",
      "GSM5025280_WT-SA1-C 100%[===================>] 391.39K  1.64MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:51:06 (1.64 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz’ saved [400787]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:06--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025281/suppl/GSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz’\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025281/suppl ... done.\n",
      "==> SIZE GSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz ... 309422\n",
      "==> EPSV ... done.    ==> RETR GSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 309422 (302K) (unauthoritative)\n",
      "\n",
      "GSM5025281_HOTTIP-K 100%[===================>] 302.17K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:07 (2.54 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz’ saved [309422]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:07--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025282/suppl/GSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025282/suppl ... done.\n",
      "==> SIZE GSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz ... 844337\n",
      "==> EPSV ... done.    ==> RETR GSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 844337 (825K) (unauthoritative)\n",
      "\n",
      "GSM5025282_WT-SA2-C 100%[===================>] 824.55K  1.88MB/s    in 0.4s    \n",
      "\n",
      "2022-02-21 22:51:08 (1.88 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz’ saved [844337]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:09--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025283/suppl/GSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025283/suppl ... done.\n",
      "==> SIZE GSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz ... 578518\n",
      "==> EPSV ... done.    ==> RETR GSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 578518 (565K) (unauthoritative)\n",
      "\n",
      "GSM5025283_HOTTIP-K 100%[===================>] 564.96K  1.40MB/s    in 0.4s    \n",
      "\n",
      "2022-02-21 22:51:10 (1.40 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz’ saved [578518]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:10--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025284/suppl/GSM5025284_captureC_HOXA-peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025284_captureC_HOXA-peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025284/suppl ... done.\n",
      "==> SIZE GSM5025284_captureC_HOXA-peaks.bed.gz ... 18818\n",
      "==> EPSV ... done.    ==> RETR GSM5025284_captureC_HOXA-peaks.bed.gz ... done.\n",
      "Length: 18818 (18K) (unauthoritative)\n",
      "\n",
      "GSM5025284_captureC 100%[===================>]  18.38K  --.-KB/s    in 0.04s   \n",
      "\n",
      "2022-02-21 22:51:11 (490 KB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025284_captureC_HOXA-peaks.bed.gz’ saved [18818]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025284_captureC_HOXA-peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:11--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025285/suppl/GSM5025285_captureC_HOTTIP-peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025285_captureC_HOTTIP-peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025285/suppl ... done.\n",
      "==> SIZE GSM5025285_captureC_HOTTIP-peaks.bed.gz ... 16530\n",
      "==> EPSV ... done.    ==> RETR GSM5025285_captureC_HOTTIP-peaks.bed.gz ... done.\n",
      "Length: 16530 (16K) (unauthoritative)\n",
      "\n",
      "GSM5025285_captureC 100%[===================>]  16.14K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:13 (111 KB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025285_captureC_HOTTIP-peaks.bed.gz’ saved [16530]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025285_captureC_HOTTIP-peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:13--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025286/suppl/GSM5025286_captureC_MECOM-peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025286_captureC_MECOM-peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025286/suppl ... done.\n",
      "==> SIZE GSM5025286_captureC_MECOM-peaks.bed.gz ... 22618\n",
      "==> EPSV ... done.    ==> RETR GSM5025286_captureC_MECOM-peaks.bed.gz ... done.\n",
      "Length: 22618 (22K) (unauthoritative)\n",
      "\n",
      "GSM5025286_captureC 100%[===================>]  22.09K   126KB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:51:14 (126 KB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025286_captureC_MECOM-peaks.bed.gz’ saved [22618]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025286_captureC_MECOM-peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:15--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025287/suppl/GSM5025287_captureC_MYC-peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025287_captureC_MYC-peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025287/suppl ... done.\n",
      "==> SIZE GSM5025287_captureC_MYC-peaks.bed.gz ... 14703\n",
      "==> EPSV ... done.    ==> RETR GSM5025287_captureC_MYC-peaks.bed.gz ... done.\n",
      "Length: 14703 (14K) (unauthoritative)\n",
      "\n",
      "GSM5025287_captureC 100%[===================>]  14.36K  --.-KB/s    in 0s      \n",
      "\n",
      "2022-02-21 22:51:16 (430 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025287_captureC_MYC-peaks.bed.gz’ saved [14703]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025287_captureC_MYC-peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:16--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025288/suppl/GSM5025288_captureC_Catenin-peaks.bed.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025288_captureC_Catenin-peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025288/suppl ... done.\n",
      "==> SIZE GSM5025288_captureC_Catenin-peaks.bed.gz ... 19939\n",
      "==> EPSV ... done.    ==> RETR GSM5025288_captureC_Catenin-peaks.bed.gz ... done.\n",
      "Length: 19939 (19K) (unauthoritative)\n",
      "\n",
      "GSM5025288_captureC 100%[===================>]  19.47K  --.-KB/s    in 0.03s   \n",
      "\n",
      "2022-02-21 22:51:17 (573 KB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025288_captureC_Catenin-peaks.bed.gz’ saved [19939]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025288_captureC_Catenin-peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:17--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025289/suppl/GSM5025289_WT-DRIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025289_WT-DRIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::10, 2607:f220:41f:250::228, 130.14.250.11, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::10|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025289/suppl ... done.\n",
      "==> SIZE GSM5025289_WT-DRIP-seq_peaks.bed.gz ... 810523\n",
      "==> EPSV ... done.    ==> RETR GSM5025289_WT-DRIP-seq_peaks.bed.gz ... done.\n",
      "Length: 810523 (792K) (unauthoritative)\n",
      "\n",
      "GSM5025289_WT-DRIP- 100%[===================>] 791.53K  4.10MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:51:18 (4.10 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025289_WT-DRIP-seq_peaks.bed.gz’ saved [810523]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025289_WT-DRIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:18--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025290/suppl/GSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::13, 2607:f220:41e:250::11, 130.14.250.12, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::13|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025290/suppl ... done.\n",
      "==> SIZE GSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz ... 736141\n",
      "==> EPSV ... done.    ==> RETR GSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz ... done.\n",
      "Length: 736141 (719K) (unauthoritative)\n",
      "\n",
      "GSM5025290_HOTTIP-K 100%[===================>] 718.89K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:19 (4.87 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz’ saved [736141]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:20--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025291/suppl/GSM5025291_WT-DRIPc-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025291_WT-DRIPc-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025291/suppl ... done.\n",
      "==> SIZE GSM5025291_WT-DRIPc-seq_peaks.bed.gz ... 396347\n",
      "==> EPSV ... done.    ==> RETR GSM5025291_WT-DRIPc-seq_peaks.bed.gz ... done.\n",
      "Length: 396347 (387K) (unauthoritative)\n",
      "\n",
      "GSM5025291_WT-DRIPc 100%[===================>] 387.06K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:20 (3.02 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025291_WT-DRIPc-seq_peaks.bed.gz’ saved [396347]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025291_WT-DRIPc-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:21--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5025nnn/GSM5025292/suppl/GSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5025nnn/GSM5025292/suppl ... done.\n",
      "==> SIZE GSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz ... 379693\n",
      "==> EPSV ... done.    ==> RETR GSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz ... done.\n",
      "Length: 379693 (371K) (unauthoritative)\n",
      "\n",
      "GSM5025292_HOTTIP-K 100%[===================>] 370.79K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:21 (2.79 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz’ saved [379693]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:22--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5081nnn/GSM5081971/suppl/GSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5081nnn/GSM5081971/suppl ... done.\n",
      "==> SIZE GSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz ... 256641\n",
      "==> EPSV ... done.    ==> RETR GSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 256641 (251K) (unauthoritative)\n",
      "\n",
      "GSM5081971_WT-LK-CT 100%[===================>] 250.63K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:23 (2.04 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz’ saved [256641]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:23--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5081nnn/GSM5081972/suppl/GSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5081nnn/GSM5081972/suppl ... done.\n",
      "==> SIZE GSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz ... 335404\n",
      "==> EPSV ... done.    ==> RETR GSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz ... done.\n",
      "Length: 335404 (328K) (unauthoritative)\n",
      "\n",
      "GSM5081972_Hottip-T 100%[===================>] 327.54K  1.98MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:51:24 (1.98 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz’ saved [335404]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:25--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5081nnn/GSM5081973/suppl/GSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5081nnn/GSM5081973/suppl ... done.\n",
      "==> SIZE GSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz ... 42634\n",
      "==> EPSV ... done.    ==> RETR GSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz ... done.\n",
      "Length: 42634 (42K) (unauthoritative)\n",
      "\n",
      "GSM5081973_WT-LK-Ch 100%[===================>]  41.63K  --.-KB/s    in 0.05s   \n",
      "\n",
      "2022-02-21 22:51:25 (896 KB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz’ saved [42634]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:26--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5081nnn/GSM5081974/suppl/GSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5081nnn/GSM5081974/suppl ... done.\n",
      "==> SIZE GSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz ... 45073\n",
      "==> EPSV ... done.    ==> RETR GSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz ... done.\n",
      "Length: 45073 (44K) (unauthoritative)\n",
      "\n",
      "GSM5081974_Hottip-T 100%[===================>]  44.02K  --.-KB/s    in 0.07s   \n",
      "\n",
      "2022-02-21 22:51:27 (647 KB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz’ saved [45073]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:28--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5529nnn/GSM5529634/suppl/GSM5529634_WT-GRO-seq_1_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529634_WT-GRO-seq_1_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5529nnn/GSM5529634/suppl ... done.\n",
      "==> SIZE GSM5529634_WT-GRO-seq_1_peaks.bed.gz ... 236133\n",
      "==> EPSV ... done.    ==> RETR GSM5529634_WT-GRO-seq_1_peaks.bed.gz ... done.\n",
      "Length: 236133 (231K) (unauthoritative)\n",
      "\n",
      "GSM5529634_WT-GRO-s 100%[===================>] 230.60K  1.34MB/s    in 0.2s    \n",
      "\n",
      "2022-02-21 22:51:28 (1.34 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529634_WT-GRO-seq_1_peaks.bed.gz’ saved [236133]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529634_WT-GRO-seq_1_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:29--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5529nnn/GSM5529635/suppl/GSM5529635_WT-GRO-seq_2_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529635_WT-GRO-seq_2_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5529nnn/GSM5529635/suppl ... done.\n",
      "==> SIZE GSM5529635_WT-GRO-seq_2_peaks.bed.gz ... 153153\n",
      "==> EPSV ... done.    ==> RETR GSM5529635_WT-GRO-seq_2_peaks.bed.gz ... done.\n",
      "Length: 153153 (150K) (unauthoritative)\n",
      "\n",
      "GSM5529635_WT-GRO-s 100%[===================>] 149.56K  --.-KB/s    in 0.09s   \n",
      "\n",
      "2022-02-21 22:51:30 (1.61 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529635_WT-GRO-seq_2_peaks.bed.gz’ saved [153153]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529635_WT-GRO-seq_2_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:30--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5529nnn/GSM5529636/suppl/GSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5529nnn/GSM5529636/suppl ... done.\n",
      "==> SIZE GSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz ... 241066\n",
      "==> EPSV ... done.    ==> RETR GSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz ... done.\n",
      "Length: 241066 (235K) (unauthoritative)\n",
      "\n",
      "GSM5529636_HOTTIP-K 100%[===================>] 235.42K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:31 (1.57 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz’ saved [241066]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "\u001b[38;5;242m\n",
      "--2022-02-21 22:51:32--  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5529nnn/GSM5529637/suppl/GSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz\n",
      "           => ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::11, 2607:f220:41e:250::13, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /geo/samples/GSM5529nnn/GSM5529637/suppl ... done.\n",
      "==> SIZE GSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz ... 206773\n",
      "==> EPSV ... done.    ==> RETR GSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz ... done.\n",
      "Length: 206773 (202K) (unauthoritative)\n",
      "\n",
      "GSM5529637_HOTTIP-K 100%[===================>] 201.93K  --.-KB/s    in 0.1s    \n",
      "\n",
      "2022-02-21 22:51:32 (1.60 MB/s) - ‘/home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz’ saved [206773]\n",
      "\n",
      "\u001b[38;5;242m0\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/geo/GSE165049/GSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz has been downloaded successfully\u001b[0m\n",
      "Finished processing 1 accession(s)\n",
      "Expanding metadata list...\n",
      "Unifying and saving of metadata... \n",
      "\u001b[92mFile /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_sample_processed.csv has been saved successfully\u001b[0m\n",
      "  Config file: /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_sample_processed.yaml\n"
     ]
    }
   ],
   "source": [
    "geofetch -i GSE165049 --processed -n bright_test --filter \".Bed.gz$\" --stored-in samples \\\n",
    "--geo-folder /home/bnt4me/Virginia/for_docs/geo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now lets list the folder to see what data is there. And let's see what's in pep files now."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31mGSM5025270_CBS-u2-KO-H3K4me3-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025271_CBS-u2-RH-H3K4me3-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025272_MYC-3CBS-RH-H3K4me3-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025273_CBS-u2-KO-H3K27me3-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025274_CBS-u2-RH-H3K27me3-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025275_MYC-3CBS-RH-H3K27me3-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025276_WT-CTCF-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025277_HOTTIP-KO-CTCF-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025278_WT-RAD21-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025279_HOTTIP-KO-RAD21-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025280_WT-SA1-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025281_HOTTIP-KO-SA1-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025282_WT-SA2-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025283_HOTTIP-KO-SA2-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025284_captureC_HOXA-peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025285_captureC_HOTTIP-peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025286_captureC_MECOM-peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025287_captureC_MYC-peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025288_captureC_Catenin-peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025289_WT-DRIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025290_HOTTIP-KO-DRIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025291_WT-DRIPc-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5025292_HOTTIP-KO-DRIPc-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5081971_WT-LK-CTCF-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5081972_Hottip-Tg-LK-CTCF-ChIP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5081973_WT-LK-ChIRP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5081974_Hottip-Tg-LK-ChIRP-seq_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5529634_WT-GRO-seq_1_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5529635_WT-GRO-seq_2_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5529636_HOTTIP-KO-GRO-seq_1_peaks.bed.gz\u001b[0m\n",
      "\u001b[01;31mGSM5529637_HOTTIP-KO-GRO-seq_2_peaks.bed.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls /home/bnt4me/Virginia/for_docs/geo/GSE165049"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSE,Sample_title,Sample_geo_accession,Sample_status,Sample_submission_date,Sample_last_update_date,S\n",
      "GSE165049,CBS-u2-KO-H3K4me3-ChIP-seq,GSM5025270,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,\n",
      "GSE165049,CBS-u2-RH-H3K4me3-ChIP-seq,GSM5025271,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,\n",
      "GSE165049,MYC-3CBS-RH-H3K4me3-ChIP-seq,GSM5025272,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,\n",
      "GSE165049,CBS-u2-KO-H3K27me3-ChIP-seq,GSM5025273,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1\n",
      "GSE165049,CBS-u2-RH-H3K27me3-ChIP-seq,GSM5025274,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1\n",
      "GSE165049,MYC-3CBS-RH-H3K27me3-ChIP-seq,GSM5025275,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA\n",
      "GSE165049,WT-CTCF-ChIP-seq,GSM5025276,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 leu\n",
      "GSE165049,HOTTIP-KO-CTCF-ChIP-seq,GSM5025277,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOL\n",
      "GSE165049,WT-RAD21-ChIP-seq,GSM5025278,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 le\n",
      "GSE165049,HOTTIP-KO-RAD21-ChIP-seq,GSM5025279,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MO\n",
      "GSE165049,WT-SA1-ChIP-seq,GSM5025280,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 leuk\n",
      "GSE165049,HOTTIP-KO-SA1-ChIP-seq,GSM5025281,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM\n",
      "GSE165049,WT-SA2-ChIP-seq,GSM5025282,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 leuk\n",
      "GSE165049,HOTTIP-KO-SA2-ChIP-seq,GSM5025283,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM\n",
      "GSE165049,WT-NG-Capture-C-seq,GSM5025284,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 \n",
      "GSE165049,HOTTIP-KO-NG-Capture-C-seq,GSM5025285,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,\n",
      "GSE165049,CBS-u2-KO--NG-Capture-C-seq,GSM5025286,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1\n",
      "GSE165049,CBS-u2-RH-NG-Capture-C-seq,GSM5025287,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,\n",
      "GSE165049,MYC-3CBS-RH-NG-Capture-C-seq,GSM5025288,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,\n",
      "GSE165049,WT-DRIP-seq,GSM5025289,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 leukemia\n",
      "GSE165049,HOTTIP-KO-DRIP-seq,GSM5025290,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 l\n",
      "GSE165049,WT-DRIPc-seq,GSM5025291,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 leukemi\n",
      "GSE165049,HOTTIP-KO-DRIPc-seq,GSM5025292,Public on Feb 18 2022,Jan 18 2021,Feb 18 2022,SRA,1,MOLM13 \n",
      "GSE165049,WT-LK-CTCF-ChIP-seq,GSM5081971,Public on Feb 18 2022,Feb 13 2021,Feb 18 2022,SRA,1,Bone ma\n",
      "GSE165049,Hottip-Tg-LK-CTCF-ChIP-seq,GSM5081972,Public on Feb 18 2022,Feb 13 2021,Feb 18 2022,SRA,1,\n",
      "GSE165049,WT-LK-ChIRP-seq,GSM5081973,Public on Feb 18 2022,Feb 13 2021,Feb 18 2022,SRA,1,Bone marrow\n",
      "GSE165049,Hottip-Tg-LK-ChIRP-seq,GSM5081974,Public on Feb 18 2022,Feb 13 2021,Feb 18 2022,SRA,1,Bone\n",
      "GSE165049,WT-GRO-seq_1,GSM5529634,Public on Feb 18 2022,Aug 19 2021,Feb 18 2022,SRA,1,MOLM13 leukemi\n",
      "GSE165049,WT-GRO-seq_2,GSM5529635,Public on Feb 18 2022,Aug 19 2021,Feb 18 2022,SRA,1,MOLM13 leukemi\n",
      "GSE165049,HOTTIP-KO-GRO-seq_1,GSM5529636,Public on Feb 18 2022,Aug 19 2021,Feb 18 2022,SRA,1,MOLM13 \n",
      "GSE165049,HOTTIP-KO-GRO-seq_2,GSM5529637,Public on Feb 18 2022,Aug 19 2021,Feb 18 2022,SRA,1,MOLM13 \n"
     ]
    }
   ],
   "source": [
    "cut -c -100 bright_test/bright_test_annotation_sample_processed.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pep_version: 2.0.0\n",
      "project_name: bright_test\n",
      "sample_table: /home/bnt4me/Virginia/for_docs/bright_test/bright_test_annotation_sample_processed.csv\n",
      "\n",
      "sample_modifiers:\n",
      "  append:\n",
      "    output_file_path: FILES\n",
      "  derive:\n",
      "    attributes: [output_file_path]\n",
      "    sources:\n",
      "      FILES: /home/bnt4me/Virginia/for_docs/geo/{SRA}/{sample_name}\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat bright_test/bright_test_annotation_sample_processed.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we have easy access to this data by using [peppy](http://peppy.databio.org/en/latest/) package in python or [pepr](https://code.databio.org/pepr/) in r in further analysis "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
